src/graph.rs

```diff
@@ -949,11 +1049,30 @@ pub(crate) fn parse_module(
   specifier: &ModuleSpecifier,
   maybe_headers: Option<&HashMap<String, String>>,
   content: Arc<String>,
+  maybe_assert_type: Option<&str>,
   maybe_resolver: Option<&dyn Resolver>,
   source_parser: &dyn SourceParser,
   is_root: bool,
 ) -> ModuleSlot {
   let media_type = get_media_type(specifier, maybe_headers);
+  // here we check any known import assertions and add them to the graph,
+  // otherwise we just continue
+  match maybe_assert_type {
+    Some("json") if media_type == MediaType::Json => {
+      return ModuleSlot::Module(Module::Synthetic(Box::new(
+        SyntheticModule::new(
+          specifier.clone(),
+          MediaType::Json,
+          None,
+          Some(content),
```

Comment:
Why are JSON modules synthetic? Synthetic modules were previously side roots (e.g. config file) that brought in ambient deps like JSX import sources.

Reply:
It has been refactored, there was no really distinction between the two and V8 creates them as "synthetic modules".  Now synthetic modules play the role of all modules that aren't ES modules, injected or otherwise, and the structure handles that they may or may not have source content.

Reply:
Why not unify them with `Module` which could may or may not have a `parsed_source`? That seems a lot more sane. TS modules are also a special variant that need extra processing by the runtime, it doesn't make sense for those to be closer to ES modules than JSON modules are.

Reply:
Integrating it into deno_doc and cli I am confident that we should keep JSON modules (and other asserted modules) as synthetic modules.  Injected sources modules are another class of synthetic modules.

There are clearly going to be refactors after this, like how we deal with JSON modules in bundling.  The fact they can "reside" in the graph, even if we refactor it later, is what we need to do for now.